api/src/main/java/pl/touk/widerest/api/common/CatalogUtils.java

```diff
@@ -17,10 +22,25 @@
 
 public class CatalogUtils {
 
-    public static Predicate<Product> nonArchivedProduct = product -> ((Status)product).getArchived() == 'N';
-    public static Predicate<Category> nonArchivedCategory = category -> ((Status)category).getArchived() == 'N';
-    public static Predicate<Product> activeProduct = Product::isActive;
-    public static Predicate<Category> activeCategory = Category::isActive;
+    public static <T> Predicate<T> isNotArchived() {
+        return ((Predicate<T>) object -> !(object instanceof Status))
+                .or(object -> ((Status) object).getArchived() != 'Y');
+    }
+
+    public static <T> Predicate<T> isUserAthorizedFor(String role) {
+        return ignored -> Optional.ofNullable(SecurityContextHolder.getContext())
+                .map(SecurityContext::getAuthentication)
+                .map(Authentication::getAuthorities)
+                .map(Collection::stream)
+                .flatMap(stream -> stream.map(GrantedAuthority::getAuthority).filter(role::equals).findAny())
+                .isPresent();
+    }
+
+    public static Predicate<Product> shouldProductBeVisible =
+        ((Predicate<Product>) Product::isActive).or(isNotArchived().and(isUserAthorizedFor("PERMISSION_READ_PRODUCT")));
+
+    public static Predicate<Category> shouldCategoryBeVisible =
```

Comment:
\[Checkstyle\] ERROR: Variable 'shouldCategoryBeVisible' must be private and have accessor methods\.